Assets/Scripts/Managers/WeaponManager.cs

```diff
@@ -75,18 +81,50 @@ public void GiveAmmo(int number)
         BulletsInInventory += number;
         HuntingUIManager.Instance.SetAmmoCounterText(BulletsInMag / currentGun.GunSO.bulletsPerTap + " / " + BulletsInInventory / currentGun.GunSO.bulletsPerTap);
     }
-
+    
+    // reloads current gun, interacts with UI
     public void Reload()
     {
-        // if there are no bullets left, don't reload
+        /*// if there are no bullets left, don't reload
         if (BulletsInInventory <= 0) return;
 
         int bulletsToReload = currentGun.GunSO.magSize - BulletsInMag;
         if (bulletsToReload > BulletsInInventory) bulletsToReload = BulletsInInventory;
 
         ownedGuns[currentGunIndex].bulletsInMag += bulletsToReload;
-        BulletsInInventory -= bulletsToReload;
+        BulletsInInventory -= bulletsToReload;*/
+
+        ReloadNoUI(currentGun);
 
         HuntingUIManager.Instance.SetAmmoCounterText(BulletsInMag / currentGun.GunSO.bulletsPerTap +  " / " + BulletsInInventory / currentGun.GunSO.bulletsPerTap);
     }
+
+    // copy of old reload function but takes a gun as a param and skips anything to do with UI.
+    private void ReloadNoUI(Gun gun)
+    {
+        if (gun == null) return;
+
+        int bulletsInv = GameManager.Instance.PermanentInventory.BoughtItems.Find(x => x.id == gun.GunSO.bulletShopItem.id).quantity;
+        int mag = ownedGuns.Find(x => x.id == gun.GunSO.id).bulletsInMag;
+
+        // if there are no bullets left, don't reload
+        if (bulletsInv <= 0) return;
+
+        int bulletsToReload = gun.GunSO.magSize - mag;
+        if (bulletsToReload > bulletsInv) bulletsToReload = bulletsInv;
```

Comment:
Recommendation: Use Mathf.Min instead
```suggestion
        int bulletsToReload = Mathf.Min(gun.GunSO.magSize - mag, bulletsInv);
```